content/doc/developer/handling-requests/json.adoc

```diff
@@ -0,0 +1,215 @@
+layout: developersection
+references:
+- url: https://github.com/stapler/stapler/blob/master/docs/reference.adoc
+  title: Stapler URL Binding Reference
+- url: ../routing
+  title: How requests in Jenkins are routed
+- url: ../../testing
+  title: Testing Jenkins
+---
+
+## Handle request in Jenkins
+
+### Pre-requisite : a POJO (Plain Old Java Object)
+
+For the exemple we are using a very simple Object.
+Note that if you use Stapler for marshalling and unmarshalling Json, you will need an empty constructor.
+
+```
+public static class MyJsonObject {
+    private String message;
+
+    //empty constructor required for JSON parsing.
+    public MyJsonObject() {}
+
+    public MyJsonObject(String message) {
+        this.message = message;
+    }
+
+    public void setMessage(String message) {
+        this.message = message;
+    }
+
+    public String getMessage() {
+        return message;
+    }
+}
+```
+
+### Handling GET with Jenkins
+
+```java
+/* (1) */
+import hudson.model.RootAction;
+import net.sf.json.JSONObject;
+import org.junit.Rule;
+import org.junit.Test;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.WebMethod;
+import org.kohsuke.stapler.json.JsonBody;
+import org.kohsuke.stapler.json.JsonHttpResponse;
+import org.kohsuke.stapler.verb.GET;
+import org.kohsuke.stapler.verb.POST;
+
+
+@Extension /* (2) */
+public static class JsonAPI implements RootAction /* (3) */ {
+
+    @CheckForNull
+    @Override
+    public String getIconFileName() {
+        return null; /* (4) */
+    }
+
+    @CheckForNull
+    @Override
+    public String getDisplayName() {
+        return null; /* (5) */
+    }
+
+    @Override
+    public String getUrlName() {
+        return "custom-api"; /* (6) */
+    }
+
+    @GET
+    @WebMethod(name = "get-example")/* (7) */
+    public /* (8) */ JsonHttpResponse getExample() {
+        JSONObject response = JSONObject.fromObject(new MyJsonObject("I am Jenkins"));
+        return new JsonHttpResponse(response, 200); /* (9) */
+    }
+
+    @GET
+    @WebMethod(name = "get-example-param")
+    public JsonHttpResponse getWithParameters(@QueryParameter(required = true) String paramValue /* (9) */) {
+        assertNotNull(paramValue);
+        JSONObject response = JSONObject.fromObject(new MyJsonObject("I am Jenkins " + paramValue));
+        return new JsonHttpResponse(response, 200);
+    }
+
+    @GET
+    @WebMethod(name = "get-error500")
+    public JsonHttpResponse getError500() { /* (10) */
+        JsonHttpResponse error500 = new JsonHttpResponse(JSONObject.fromObject(new MyJsonObject("You got an error 500")), 500);
+        throw error500;
+    }
+}
+```
+1. Non exhaustive list of imports to use.
+2. Need to be an extension to be discovered as a service by Jenkins.
+3. This class is an extension of RootAction, it's a way to expose HTTP paths. It's more frequently use to expose Web UI, but it can be use also without HTML rendering.
```

Comment:
```suggestion
3. This class is an extension of RootAction.  It is a way to expose HTTP paths that is more frequently used to expose the Web UI, but it can also be used without HTML rendering.
```